CRM/Civicase/APIHelpers/CustomGroups.php

```diff
@@ -48,4 +48,50 @@ public static function getIdForGroupName($customGroupName) {
     return NULL;
   }
 
+  /**
+   * Activates a list of custom groups for the given entity.
+   *
+   * This shows them in UI too.
+   *
+   * @param string $entityName
+   *   The name of the entity.
+   *
+   * @return array
+   *   Custom Group Api response.
+   */
+  public static function activateCustomGroupsForEntity($entityName) {
+    return civicrm_api3('CustomGroup', 'get', [
+      'extends' => $entityName ,
+      'options' => ['limit' => 0],
+      'return' => 'id',
+      'api.custom_group.create' => [
+        'id' => '$value.id',
+        'is_reserved' => FALSE,
```

Comment:
Not sure what you are trying to do in this function and in the `deactivateCustomGroupsForEntity` function but setting a custom group to reserved only hides it from the UI but the custom fields for these field sets will still be active on the respective entities.
If we want to activate or deactivate it should be via the `is_active` field by setting to TRUE or FALSE.